src/org/sosy_lab/java_smt/solvers/z3/Z3Model.java

```diff
@@ -206,7 +208,8 @@ private Collection<ValueAssignment> getConstantArrayAssignment(
    * @return a list of assignments {@code a[1]=0; a[2]=0; a[5]=0}.
    */
   private Collection<ValueAssignment> getArrayAssignments(
-      long arraySymbol, long arrayFormula, long value, List<Object> upperIndices) {
+      long arraySymbol, long arrayFormula, long value, List<Object> upperIndices)
+      throws InterruptedException {
     long evalDecl = Native.getAsArrayFuncDecl(z3context, value);
     Native.incRef(z3context, evalDecl);
     long interp = Native.modelGetFuncInterp(z3context, model, evalDecl);
```

Comment:
Based on the annotations from https://github.com/Z3Prover/z3/blob/master/src/api/java/Model.java , it seems that methods like "modelGetFuncInterp" can also throw a Z3Exception. We might want to check more places for InterruptedException.

Reply:
fixed